compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -3117,16 +3117,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     }
                 }
                 ty::Tuple(tys) => {
-                    let fstr = field.as_str();
-
-                    if let Ok(index) = fstr.parse::<usize>() {
-                        if fstr == index.to_string() {
-                            if let Some(&field_ty) = tys.get(index) {
-                                field_indices.push(index.into());
-                                current_container = field_ty;
+                    if let Ok(index) = field.as_str().parse::<usize>()
+                        && field.name == sym::integer(index)
+                    {
+                        for ty in tys.iter().take(index + 1) {
+                            self.require_type_is_sized(ty, expr.span, traits::MiscObligation);
+                        }
```

Comment:
What is the reason for checking *all* fields of the tuple (that are before the indexed field)? For ADTs we only check the field we are indexing, and the same should be enough for tuples. So I don't see the point of this loop.

In https://github.com/rust-lang/rust/pull/126150 I am removing it.

Reply:
At this point I can't remember 🤔 I think all good if it's caught by code elsewhere